File: Prudhvi-LLD-Mar/BehavioralDesignPattern/WeatherStationObserverDesignPattern/StatisticsDisplay.java
package WeatherStationObserverDesignPattern;


import java.text.MessageFormat;

public class StatisticsDisplay implements Display {

  private double currentTemperature;
  private double currentHumidity;

  @Override
  public void update(WeatherStationMessage weatherStationMessage) {
    this.currentTemperature = weatherStationMessage.getTemperature();
    this.currentHumidity = weatherStationMessage.getHumidity();
  }

  @Override
  public String display() {
    return MessageFormat.format("Statistics: Temperature: {0}°C, Humidity: {1}%",
        currentTemperature, currentHumidity);
  }
}

File: rahul-narang-LLD-March/src/StatisticsDisplay.kt
class StatisticsDisplay(
    var name: String
) : Observer {

    private var temp: Double = 0.0
    private var humidity: Double = 0.0
    private var pressure: Double = 0.0

    override fun update(weatherStationMessage: WeatherStationMessage) {
        this.temp = weatherStationMessage.temp
        this.humidity = weatherStationMessage.humidity
        this.pressure = weatherStationMessage.pressure
    }

    fun getCurrentWeatherConditions(): String {
        return "Statistics: Temp: ${this.temp} degree C, Humidity: ${this.humidity} %, Pressure: ${this.pressure} hpa"
    }
}